### DISCUSSION TITLE
Funding rate might confuse people

### DISCUSSION THREAD

**Discussion by bonomat** [get10101/itchysats#1751] [General]
The way how funding rate is depicted at the moment might be confusing. 
I know that a positive funding rate means, long pays short but this is not really visible nor described anywhere. 
Right now, it looks like that both, longs and shorts will need to pay this rate: 

![image](https://user-images.githubusercontent.com/224613/160261374-e674a076-a49a-485a-a049-e149e22c0993.png)

![image](https://user-images.githubusercontent.com/224613/160261378-f118b0a8-89fc-4d11-be90-f19b85368a6a.png)

**Comment by klochowicz** [comment]
isn't this the case in the above pictures that both long and short need to pay, ie. it's not a negative funding rate? (just double-checking as the rates are quite different in long and short tab)

**Reply by bonomat** [reply]
mhm... good that it even confuses us 😅 Imho it means: positive means long pay, negative means short pay. So in this case, when shorting, and it says 0.00026, it means that the shorts would receive this amount.

**Reply by klochowicz** [reply]
I don't think this would be a good UX for the end-user if we expose our internal data model without translating. It makes a lot of sense internally, but it would be extremely hard for the end user to realise that. IMHO it would be way easier for the end user to always be negative when *they* need to pay, and positive if they receive compensation.

**Reply by bonomat** [reply]
Totally agree with you :) 
I just stated how it is implemented at the moment

**Comment by bonomat** [comment]
> isn't this the case in the above pictures that both long and short need to pay, ie. it's not a negative funding rate? (just double-checking as the rates are quite different in long and short tab)

mhm... good that it even confuses us 😅 Imho it means: positive means long pay, negative means short pay. So in this case, when shorting, and it says 0.00026, it means that the shorts would receive this amount.

**Comment by thomaseizinger** [comment]
It appears to me that we should fix this in the daemon and return fields like:

```json
{
	"you_pay_hourly": "0.000026%"
}
```
and

```json
{
	"you_receive_hourly": "0.000026%"
}
```

Actual names subject to bike-shedding. The main point being: The daemon knows, which role it is in and thus, could map this directly to an unambiguous representation. I would suggest to have a think about what we want to display in the UI and then make the API resemble this as closely as possible.

**Comment by thomaseizinger** [comment]
Meta: Moved this to a discussion because it is not yet actionable.

**Comment by klochowicz** [comment]
this discussion is over, we agreed on solution briefly outlined in #1877.

**Comment by bonomat** [comment]
I propose to move away from a `funding rate` and use `interest rate` instead.

## Context

Funding rate is a value which indicates the market's sentiment because the spot market price and the perpetual contracts price diverge.  i.e. if it is positive, the market is more bullish, if it is negative, the market is bearish.
The funding rate flows from either long to short or short to long and depends on the market sentiment, i.e:

- if it is positive, long pay short
- if it is negative, short pay long

This can lead to an incentive to keep the position open or close it.

The problem with the funding rate for us is two-fold:

1. On centralized exchanges there is usual **one** funding rate for the whole exchange which is decided by "The market".
   We can't have this easily as there is no central party under control.
   Instead, as of now (and probably for a long time) the maker has to come up with the funding rate.

Note: "The market" can still regulate the rate: if a rate offered by a maker does not fit to the market sentiment, no one will take his offers.

2. A maker offers a short and long position. Given the maker's strategy the rates offered for longs and shorts might be different, leading to two different _funding rates_.

In both cases, the _funding rate_ needs to be paid or is paid out as long as the position is open.

This sounds more like _interest rates_: if you have borrow something from the bank (i.e. your position is open), you either pay interest from to the bank, or, if you give something to the bank, you receive interest from the bank.

## Proposal

Move to interest rates:
   The ticket above says the funding rate should be reworked to indicate wether a taker has to pay or would receive something.
   This could be a purely frontend solution but we still have the problem of having two different _funding rates_.

Additionally, only changing this in the frontend can lead to confusion for developers, because the term _funding rate_ is deeply integrated into our daemon and database. It can be changed but will take some time as services built on top also makes use of this logic.

In the long run I would see a split between the different assets and provide two different interest rates: 
- interest on the base currency
- interest on the quote currency
The difference between both determines the interest rate you have to pay (or get paid) to your counterparty.

**Comment by bonomat** [comment]
This ticket is related: #2107 

Based on the function in 

https://github.com/itchysats/itchysats/blob/20bfc556eebfcf9e55a617ef78b0cc72ded16e6e/model/src/lib.rs#L691-L733 

Again I claim that our funding rate is an interest rate and not a funding rate, i.e. 
- you pay if you borrow funds from someone, or
- you receive money if you borrow something to others.

**Reply by klochowicz** [reply]
can't we agree that funding rate can be the same?
- you pay if someone else funds
- you get paid if you fund

"funding" above means putting up money against market sentiment.

I wouldn't say it's interest rate, because that one would be clearly going to the person locking up more money - and that is not always the case.

**Reply by bonomat** [reply]
> [..snip] because that one would be clearly going to the person locking up more money - and that is not always the case.

Who says that?  
The bank clearly has more money locked up than you, yet, on your saving accounts, you get interest paid out in the interest rates are positive.

> "funding" above means putting up money against market sentiment.

According to centralized exchanges: the funding rate is based on the traded quantity not on the amount you lock up. 
Or differently said in words from Bitmex: 

> The funding you pay or receive is calculated as:
>
>	`Funding = Mark Value * Funding Rate`
>
> Your Mark Value is irrespective of leverage. For example, if you hold 100 XBTUSD contracts, funding is charged/received on the notional value of those contracts, and is not based on how much margin you have assigned to the position.

Source: https://www.bitmex.com/app/perpetualContractsGuide#Funding

For us, our our wrongly named _funding rate_ is based on the other party's margin or your own margin depending on whether the rates are positive or negative?

**Reply by klochowicz** [reply]
yet in the mortgage case, the interest rates are clearly positive and yet you pay to the bank. I perceive interest rates to be inherently combined with the capital someone has put up - either the person for the bank, or the bank for them. 

Perhaps I should have been more explicit in my example above  - the bank does not have more money locked up than you in the savings account, it has more money in general (which is irrelevant).

I still can't see how renaming this term will make the users confused less, IMHO we will end up with *the same problem*, and the following discussion will be "Interest rate might confuse people".  This issue/discussion/observation was due to the fact that we thought we should show negative values when the user *pays* and positive when he *is paid*. Are we on the same page with this, irrespectively of how we name the rate?

**Reply by bonomat** [reply]
> Perhaps I should have been more explicit in my example above - the bank does not have more money locked up than you in the savings account, it has more money in general (which is irrelevant).

Yea, let's ignore examples outside of our field and generalization like `if interest rates are positive, it always means that you have to pay` because this is not true. The direction of payments based on interest rates depends on the financial product you are using. 

> I still can't see how renaming this term will make the users confused less, IMHO we will end up with the same problem, and the following discussion will be "Interest rate might confuse people". 

Obviously, understanding of something is always subjective and you will always find people who will not understand something.
I would argue that most people understand interest rate, while funding rate is something new. 

I think it confuses us (the developers) which is a bigger problem and will be the source of bugs.

> This issue/discussion/observation was due to the fact that we thought we should show negative values when the user pays and positive when he is paid. Are we on the same page with this, irrespectively of how we name the rate?

That's how it started. I then realized that our _funding rate_ is not a funding rate but an _interest rate_ and hence started the discussion. 

I will point it out again in case it got lost: if we use the wording _funding rate_ we will confuse people and ourselves because it **does not** behave like a _funding rate_ on other exchanges.

**Reply by klochowicz** [reply]
OK, we might have 2 distinct discussions running in a single one, and that's the reason we might be spinning around:
1. Interest rate is a better term for what we use in ItchySats
2. Should we use negative and positive numbers to the end user

I think 1) is irrelevant to 2)'s resolution, unless it somehow becomes clearer? Perhaps you could provide examples from the UI on how it becomes clearer to solve 2)?

I took the assumption that people != devs, and that this was about communicating fees to the user. We don't seem to be confused so far by how we calculate and apply funding fees so far, or am I mistaken?


I still don't know whether we're on the same page about the *sign* of whatever we present to the user, irrespective of how we call it. I care about 2), the naming can be changed and defined by us in our FAQ as we exactly want. For the record, I am not fundamentally opposed to the new term, I just don't see it fixing the underlying issue in this discussion (I see how it can solve some issues like distinguishing from bitmex terms to not confuse us, and us such it might be worth it on its own).

**Reply by bonomat** [reply]
For me these two values are depending on each other for both terms and cannot be treated separately.

The advantage of the `interest rate` is that it is user specific, i.e. the user knows exactly what he will get or has to pay. 
Compared to `funding rate` where the user has to put his position into perspective to understand whether he will get something or has to pay something.

**Reply by luckysori** [reply]
> The advantage of the interest rate is that it is user specific, i.e. the user knows exactly what he will get or has to pay.
Compared to funding rate where the user has to put his position into perspective to understand whether he will get something or has to pay something.

I thought it would be absolutely valid to:

- Display the global funding rate so that a user can understand what kind of position would costs/earns what.
- Display a per-position funding rate that simply shows whether they're paying or earning for that position.
